src/vma/dev/allocator.cpp

```diff
@@ -234,6 +234,10 @@ bool vma_allocator::register_memory(size_t size, ib_ctx_handler *p_ib_ctx_h,
 vma_allocator::~vma_allocator() {
 	// Unregister memory
 	for (size_t i = 0; i < m_mr_list_len; ++i) {
+		if (!m_mr_list[i]) {
+			__log_info_err("ibv_mr is null, list length is %zd", m_mr_list_len);
+			continue;
+		}
```

Comment:
It does not make sense to have a NULL in mr_list.
A better approach would be not to add NULL element to the list when the registration failed.
Another weak spot :
`uint32_t vma_allocator::find_lkey_by_ib_ctx(ib_ctx_handler *p_ib_ctx_h) const
{
        ibv_device* dev = p_ib_ctx_h->get_ibv_device();
        for (size_t i = 0; i < m_mr_list_len; ++i) {
                if (dev == m_mr_list[i]->context->device) { <---- crash
                        return m_mr_list[i]->lkey;
                }
        }
        return 0;
}
`
Let's talk about it on Sunday.

Reply:
this flow will never happen due to the fact that reg_mr fails so VMA exits.
but i handled the cause of the bug (bad array size setting) in this new commit